common/channel/src/libra_channel.rs

```diff
@@ -0,0 +1,151 @@
+use std::collections::{HashMap, VecDeque};
+use std::pin::Pin;
+use std::sync::{Arc, Mutex};
+use std::task::Waker;
+
+use futures::async_await::FusedStream;
+use futures::stream::Stream;
+use futures::task::Context;
+use futures::Poll;
+
+/// ValidatorMessage trait needs to be implemented by every message which gets pushed into the Libra
+/// channel so that we can ensure fairness among validators
+pub trait ValidatorMessage {
+    /// Extract the Validator from which this message arrived. This
+    /// will be used for ensuring fairness among validators.
+    fn get_validator(&self) -> &str;
```

Comment:
minor: I don't think we want a string here. 
Today we're using a struct `AccountAddress` for peer ids. So we could either use a template parameter, which implements `Eq`, `Hash` (in this case the channel is going to provide fairness based on an abstract `Key`), or a specific struct that we can use for validators. @phlip9, what's the ultimate struct in libra types for the validator? Is it `AccountAddress`?

Reply:
It is `AccountAddress`

Reply:
Updated to use `AccountAddress`. Let me know if you want me to update it to a generic parameter.